cmd/collector/app/zipkin/zipkin_json.go

```diff
@@ -0,0 +1,234 @@
+)
+
+type endpoint struct {
+	ServiceName string `json:"serviceName"`
+	IPv4        string `json:"ipv4"`
+	IPv6        string `json:"ipv6"`
+	Port        int16  `json:"port"`
+}
+type annotation struct {
+	Endpoint  endpoint `json:"endpoint"`
+	Value     string   `json:"value"`
+	Timestamp int64    `json:"timestamp"`
+}
+type binaryAnnotation struct {
+	Endpoint endpoint `json:"endpoint"`
+	Key      string   `json:"key"`
+	Value    string   `json:"value"`
+}
+type zipkinSpan struct {
+	ID                string             `json:"id"`
+	ParentID          string             `json:"parentId,omitempty"`
+	TraceID           string             `json:"traceId"`
+	Name              string             `json:"name"`
+	Timestamp         *int64             `json:"timestamp"`
+	Duration          *int64             `json:"duration"`
+	Debug             bool               `json:"debug"`
+	Annotations       []annotation       `json:"annotations"`
+	BinaryAnnotations []binaryAnnotation `json:"binaryAnnotations"`
+}
+
+var (
+	errIstUnsignedLog = errors.New("id is not an unsigned long")
+)
+
+func deserializeJSON(body []byte) ([]*zipkincore.Span, error) {
+	spans, err := decode(body)
+	if err != nil {
+		return nil, err
+	}
+
+	return spansToThrift(spans)
+}
+
+func decode(body []byte) ([]zipkinSpan, error) {
+	type zipkinSpans []zipkinSpan
+	var spans zipkinSpans
+
+	if err := json.Unmarshal(body, &spans); err != nil {
+		return nil, err
+	}
+
+	return spans, nil
+}
+
+func spansToThrift(spans []zipkinSpan) ([]*zipkincore.Span, error) {
+	var tSpans []*zipkincore.Span
+	for _, span := range spans {
+		tSpan, err := spanToThrift(span)
+		if err != nil {
+			return nil, err
+		}
+		tSpans = append(tSpans, tSpan)
+	}
+	return tSpans, nil
+}
+
+func spanToThrift(span zipkinSpan) (*zipkincore.Span, error) {
+	id, err := hexToUnsignedLong(span.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	var traceID uint64
+	if len(span.TraceID) == 32 {
+		traceID, err = hexToUnsignedLong(span.TraceID[16:])
+	} else {
+		traceID, err = hexToUnsignedLong(span.TraceID)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	tSpan := &zipkincore.Span{
+		ID:        int64(id),
+		TraceID:   int64(traceID),
+		Name:      span.Name,
+		Debug:     span.Debug,
+		Timestamp: span.Timestamp,
+		Duration:  span.Duration,
+	}
+
+	if len(span.ParentID) > 0 {
+		parentID, err := hexToUnsignedLong(span.ParentID)
+		if err != nil {
+			return nil, err
+		}
+		signed := int64(parentID)
+		tSpan.ParentID = &signed
+	}
+
+	for _, anno := range span.Annotations {
+		anno, err := annoToThrift(anno)
+		if err != nil {
+			return nil, err
+		}
+		tSpan.Annotations = append(tSpan.Annotations, anno)
+	}
+	for _, binAnno := range span.BinaryAnnotations {
+		binAnno, err := binAnnoToThrift(binAnno)
+		if err != nil {
+			return nil, err
+		}
+		tSpan.BinaryAnnotations = append(tSpan.BinaryAnnotations, binAnno)
+	}
+
+	return tSpan, nil
+}
+
+func endpointToThrift(endp endpoint) (*zipkincore.Endpoint, error) {
```

Comment:
go prefers incomprehensible variable names for short functions: https://talks.golang.org/2014/names.slide#7

so here s/endp/e